### ISSUE TITLE
Allow sending raw request strings

### ISSUE THREAD

**Issue by mpociot** [clue/reactphp-buzz#158]
As discussed with @WyriHaximus:

It would be great if the browser class would allow me to not only send a PSR7 Request object but send a raw request string as well.
I am building an HTTP tunneling application and receive the raw HTTP request strings. Right now I transform those requests to PSR7 requests to send them.
This works, but it introduces an additional conversation, where things could potentially go wrong. 

Therefore it would be good if I could do something like:

```
(new Browser())
    ->sendRaw($requestString)
    ->then(...
```

**Comment by clue** [comment]
@mpociot Interesting one! I understand where you're coming from and can indeed see some use for this.

That being said, this is anything but trivial and would require us to parse the passed request anyway, plus it would cause issues with large request bodies (think file uploads etc.). Internally, we need to create a TCP/IP connection to the destination address. This *could* technically be parsed from the given `Host` request header, but this header isn't guaranteed to match the destination address (proxy setup), let alone even be present for HTTP/1.0 requests. On top of this, this would make it much harder one we support HTTP/2 in the future which uses a binary message format.

Long story short, I think passing it through our explicit APIs is probably the best way forward as is guarantees all required arguments are available.

I believe this has been answered, so I'm closing this for now. Please come back with more details if this problem persists and we can reopen this :+1:
